src/uwtools/config/formats/ini.py

```diff
@@ -13,16 +15,13 @@ class INIConfig(Config):
 
     DEPTH = 2
 
-    def __init__(
-        self,
-        config_file: str,
-    ):
+    def __init__(self, config: Union[dict, OptionalPath] = None):
```

Comment:
Permit creating `Config` objects with an initial `dict` in addition to the traditional external file. This is helpful for API uses cases where we have an in-memory dict and need a `Config` object based on it to conform to an interface requiring a `Config` argument, to e.g. to validate the config represented by the `dict`. It will be useful in near-future work allowing creation of `Config` objects via the API, specifically for cases where the user wants to start from an in-memory `dict`.

Similar support has been added in the subclasses of `Config`, below, where needed.

Another option would be to rename the `config_file` argument to `config` and extend its type to `Union[dict, OptionalFile]`, so that `Config(config={})` would obtain an empty `Config`, but `Config(config=d)` would initialize it with an arbitrary `dict`. Currently, some code (especially in tests) does `c = Config(empty=True)` followed by `c.update(d)`, and those could be combined in a single statement. Thoughts?